packages/ingest/s3.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const aws = require('@cumulus/common/aws');
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const urljoin = require('url-join');
+
+module.exports.s3Mixin = (superclass) => class extends superclass {
+
+  /**
+   * Downloads a given url and upload to a given S3 location
+   *
+   * @param {string} filepath - the locatio of the file to be uploaded
+   * @param {string} bucket - the S3 bucket to upload to
+   * @param {string} key - the base path of the S3 key
+   * @param {string} filename - the filename to be uploaded
+   * @returns {Promise} a promise
+   * @private
+   */
+  async sync(filepath, bucket, key, filename) {
```

Comment:
This function can actually be deleted, it's never actually used.

Reply:
We are using download() and upload() during ingest, and I would think it would be better to use sync() for S3 protocol.  I would leave it there for now.

Reply:
@scisco , care to weigh in?

Reply:
@jennyhliu if I remember correctly, we switched to download/upload instead of sync because there are few things that has to be done on granule files after download but before upload such as checksum validation, etc. 

I think we should either remove sync from all the mixings (`http.js`, `ftp.js`, etc.) or leave them all in there for now.